tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3959 @@
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Alias ID</td>
+                        <td>ByteArray[20]</td>
+                        <td>The raw bytes of the <i>Alias ID</i> which is the BLAKE2b-160 hash of the outputID that created it.</td>
+                    </tr>
+                </table>
+            </details>
+            <details>
+                <summary>NFT Address</summary>
+                <table>
+                    <tr>
+                        <td><b>Name</b></td>
+                        <td><b>Type</b></td>
+                        <td><b>Description</b></td>
+                    </tr>
+                    <tr>
+                        <td>Address Type</td>
+                        <td>uint8</td>
+                        <td>
+                            Set to <strong>value 16</strong> to denote an <i>NFT Address</i>.
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>NFT ID</td>
+                        <td>ByteArray[20]</td>
+                        <td>The raw bytes of the <i>NFT ID</i> which is the BLAKE2b-160 hash of the outputID that created it.</td>
+                    </tr>
+                </table>
+            </details>
+        </td>
+    </tr>
+</table>
+
+| :information_source:  Good to know about address format |
+|---------------------------------------------------------|
+
+The <i>Address Type</i> byte of a raw address has an effect on the starting character of the bech32 encoded address,
+which is the recommended address format for user facing applications.
+
+A usual bech32 encoded mainnet address starts with `iota1`, and continues with the bech32 encoded bytes of the address.
+By choosing <i>Address Type</i> as a multiple of 8 for different address types, the first character after the `1`
+separator in the bech32 address will always be different.
+
+| Address | Type Byte as `uint8` | Bech32 Encoded |
+| ------- | --------- | -------------- |
+| Ed25519 | 0 | iota1**q**... |
+| Alias | 8 | iota1**p**... |
+| NFT | 16 | iota1**z**... |
+
+A user can identify by looking at the address whether it is a signature backed address, a smart contract chain account
+or an NFT address.
+
+##### Storage Deposit Return Unlock Condition
+
+This unlock condition is employed to achieve conditional sending. An output that has
+<i>Storage Deposit Return Unlock Condition</i> specified can only be consumed in a transaction that deposits
+`Return Amount` IOTA coins into `Return Address`. When several of such outputs are consumed, their return amounts per
+`Return Addresses` are summed up and the output side must deposit this total sum per `Return Address`.
+
+###### Additional syntactic transaction validation rule:
+
+- `Return Amount` must be ≥ than `Minimum Storage Deposit` and must not be `0`.
+- It must hold true, that `0` ≤ `Amount` - `Return Amount` ≤ `Required Dust Deposit of the Output`.
```

Comment:
If I understand correctly, `Minimum Storage Deposit` and `Required Dust Deposit of the Output` are actually the same? So we could unify this a little bit:

```suggestion
- `Return Amount` must not be `0` and must be ≥ than `Minimum Storage Deposit of the Output`.
- It must hold true, that `0` ≤ `Amount` - `Return Amount` ≤ `Minimum Storage Deposit of the Output`.
```

Reply:
Thanks for clearing this up!

Reply:
> [that's why you also need to inspect the address within the condition to find out the minimum btw](https://github.com/iotaledger/iota.go/blob/3d39187c512978b898364c6b0512bb96b4730a48/output.go#L696).

But there you're missing that the input with the `StorageDepositReturnUnlockCondition` could also have native tokens, which should also be sent back and require a larger deposit then, no?

Reply:
The idea is to be able to create an output with native tokens where the `StorageDepositReturnUnlockCondition` allows you to specify the iota tokens you get back once the receiver wants to consume the output. Don't think we modelled it to support a use case where you want to have native tokens sent back up on consumption. Can you specify a scenario in which this would come into play?

Reply:
Sorry ignore my comment, I mixed it up with the expiration unlock condition